[Feature] Added options object to print API

andreysaf opened the pull request:
Added options to print API:
- All pages
- pages to print,
- include comments
- include annotations

Tested with API, print through UI, print through cmd+P using the built version.

ZhijieZhang commented on src/helpers/hotkeysManager.js:
```diff
@@ -6,7 +6,7 @@ import openFilePicker from 'helpers/openFilePicker';
 import setToolModeAndGroup from 'helpers/setToolModeAndGroup';
 import { zoomIn, zoomOut } from 'helpers/zoom';
-import { print } from 'helpers/print';
+import print from 'src/apis/print';
```
Our convention is that the internal code shouldn't use any exposed APIs.
So I think it'd be better if we still use
```
import { print } from 'helpers/print';
```
and then pass all the needed state into this helper function

ZhijieZhang commented on src/components/MenuOverlay/MenuOverlay.js:
```diff
@@ -6,11 +6,11 @@ import core from 'core';
 import downloadPdf from 'helpers/downloadPdf';
 import openFilePicker from 'helpers/openFilePicker';
-import { print } from 'helpers/print';
+import print from 'src/apis/print';
```
Same here, see comments in `hotkeysManager.js`.

ZhijieZhang commented on src/apis/print.js:
```diff
@@ -9,13 +13,32 @@ WebViewer(...)
+    allPages: false,
+    pagesToPrint: [],
+    includeAnnotations: false,
+    includeComments: false,
```
Should `allPages` be `true` and `includeAnnotations` be `true` to match the default options in the print modal?

ZhijieZhang commented on src/apis/print.js:
```diff
@@ -1,6 +1,10 @@
+ * @param {boolean} [allPages] If true, all pages will be printed
+ * @param {Array.<number>} [pagesToPrint] If allPages is false and pagesToPrint is passed in, will print only the pages passed in
+ * @param {boolean} [includeAnnotations] If true, will print the documents with the annotations
+ * @param {boolean} [includeComments] If true, will append comments to the document printed
```
I think it'd be good to document what the options will be. One way you can do this is
```
{boolean} [allPages=false]
```
or just have something like `Default is false` at the end of line

ZhijieZhang commented on src/helpers/print.js:
```diff
@@ -32,6 +35,29 @@ export const print = (dispatch, isEmbedPrintSupported) => {
       dispatch(actions.closeElement('loadingModal'));
     });
+  } else if (allPages || includeAnnotations || includeComments || pagesToPrint.length > 0) {
+    if (allPages || pagesToPrint.length === 0) {
```
I feel like it's more natural to not print any pages if users call

```
instance.print({
  allPages: false,
  includeAnnotations: true,
  pagesToPrint: []
})
```

mparizeau-pdftron commented:
What about creating a new function called something like `instance.printInBackground` that would just be doing the programmatic print, then we can leave `instance.print` as is? In the future we could consider merging them or renaming them to make more sense as part of a major release.

andreysaf commented:
@mparizeau-pdftron 

- added a new API called instance.printInBackground(options)
- left instance.print() same

@ZhijieZhang 

- added defaults to the API
- defaults are all false, or blank unless the user provides them in the API
- using helpers instead of external APIs
- added a console.warn for trying to call allPages=false and pagesToPrint=[]

ZhijieZhang commented on src/apis/printInBackground.js:
```diff
@@ -0,0 +1,49 @@
+ * @param {boolean} [allPages=false] If true, all pages will be printed
+ * @param {Array.<number>} [pagesToPrint=[]] If allPages is false and pagesToPrint is passed in, will print only the pages passed in
+ * @param {boolean} [includeAnnotations=false] If true, will print the documents with the annotations
+ * @param {boolean} [includeComments=false] If true, will append comments to the document printed
```
All the 4 options are optional but if I just call `instance.printInBackground()` I will see
```
webviewer-ui.min.js:50 No pages were passed in to be printed. Please pass either allPages set to true or a range of pages in pagesToPrint.
```
in the console. I think this essentially means that `allPages: true` or `pagesToPrint: ...` is a required option to use this API, which contradicts the documentation.

I think we should either change the documentation here or change the defaultOptions in the code.

ZhijieZhang commented on src/apis/printInBackground.js:
```diff
@@ -0,0 +1,49 @@
+/**
+ * Print the current document.
```
I suggest we explain a little bit about the difference between `printInBackground` and `print`. In a user's perspertive I think it can be confusing for seeing two different APIs with the same description.

andreysaf commented:
@ZhijieZhang added better docs, removed the check for pagesToPrint being zero, adjusted the required params
